### DISCUSSION TITLE
JSON/CSV import via a Flow

### DISCUSSION THREAD

**Discussion by dmatthams** [directus/directus#17552] [Feature Requests]
### Summary

The CSV/Json import is very basic – there is no way to map columns to database fields, or sanitise the data on import. I can however pass some json to a webhook and import/sanitise the data that way, but I need to make a seperate front-end for this when using Directus cloud. Pehaps we could have a way to send JSON/CSV files to a Flow as a JSON payload?

### Motivation

It will give us a powerful and flexible way to import data into directus without needing an extension, or creating a seperate front-end.

### Detailed Design

Potentially create an 'Import' trigger for flows which ovverides the default action of the import module in the collection's side bar. E.g. You'd set up a flow with an 'Import' trigger for that collection, then when a user uploads a CSV/Json file it gets sent to that flow as a Json payload and you can do what you want with it?

### Must Have:
- A way to send CSV/JSON files to a flow as JSON

### Should Have:
- 

### Could Have:
- A way to ovveride the default 'import' action

### Won't Have:
- 

### Drawbacks

I'm not convinced there are any 🤔

### Alternatives

The ability to read files in a flow on Cloud, perhaps even allowing you to trigger a flow from an asset.

### Adoption Strategy

No breaking changes, completely optional.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
